src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
+    private static double SIZE_OF_INCREASE = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[CAPACITY];
+    }
+
+    private void increaseCapacity() {
+        if (values.length == size) {
+            int newCapacity = (int)(values.length * SIZE_OF_INCREASE);
+            values = Arrays.copyOf(values, newCapacity);
+        }
```

Comment:
The method 'increaseCapacity' should not check if the array needs to grow inside of it. The method name should reflect that it always increases the array size. Consider renaming it to reflect its actual functionality or moving the size check to the caller method.